src/configuration.ts

```diff
@@ -510,21 +511,35 @@ export function parseConfigFile(
     }
 
     function parse(config: RawConfigFile, dir?: string): IConfigurationFile {
+        const rulesDirectory: string | string[] = getRulesDirectories(config.rulesDirectory, dir);
+        const jsRules = typeof config.jsRules === "boolean" ?
+                            parseRules(config.rules, config.jsRules, rulesDirectory) :
+                            parseRules(config.jsRules);
+
         return {
             extends: arrayify(config.extends),
-            jsRules: parseRules(config.jsRules),
+            jsRules,
             linterOptions: parseLinterOptions(config.linterOptions, dir),
             rules: parseRules(config.rules),
-            rulesDirectory: getRulesDirectories(config.rulesDirectory, dir),
+            rulesDirectory,
         };
     }
 
-    function parseRules(config: RawRulesConfig | undefined): Map<string, Partial<IOptions>> {
+    function parseRules(config: RawRulesConfig | undefined,
+                        copyRulestoJsRules = false,
+                        rulesDirectory?: string | string[]): Map<string, Partial<IOptions>> {
         const map = new Map<string, Partial<IOptions>>();
         if (config !== undefined) {
             for (const ruleName in config) {
                 if (hasOwnProperty(config, ruleName)) {
-                    map.set(ruleName, parseRuleOptions(config[ruleName], defaultSeverity));
+                    const ruleOptions = parseRuleOptions(config[ruleName], defaultSeverity);
+                    if (copyRulestoJsRules && ruleOptions.ruleSeverity !== "off") {
+                        const Rule = findRule(ruleName, rulesDirectory);
+                        if (Rule === undefined || (Rule.metadata && Rule.metadata.typescriptOnly)) {
```

Comment:
Is there a nicer way to find if a rule is typescript only? Because for this i need to pass in rulesDirectory which makes the parent parse function a bit less clean